lgc/unittests/interface/OptLevelTest.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ ***********************************************************************************************************************
+ *
+ *  Copyright (c) 2022 Google LLC. All Rights Reserved.
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *
+ **********************************************************************************************************************/
+
+#include "lgc/EnumIterator.h"
```

Comment:
nit: you can include `llvm/ADT/Sequence.h` for `enum_seq` directly. It may also make sense to mark the `CodeGenOpt::Level` safe for iteration in LLVM so that we don't have to use `force_iteration_on_noniterable_enum`.

Reply:
Or alternatively, we could list all opt levels by hand, since there are only a few we are interested in:
```c++
using namespace CodeGenOpt;
for (auto optLevel : {Level::None, ..., Level::Aggressive})
```